core/src/main/java/com/netease/arctic/utils/ArcticTableUtil.java

```diff
@@ -34,4 +46,79 @@ public static String tableRootLocation(ArcticTable arcticTable) {
     }
     return tableRootLocation;
   }
+
+  public static StructLikeMap<Long> readOptimizedSequence(KeyedTable table) {
+    return readWithLegacy(table.baseTable(), null, BLOB_TYPE_OPTIMIZED_SEQUENCE);
+  }
+
+  public static StructLikeMap<Long> readOptimizedSequence(KeyedTable table, long snapshotId) {
+    return readWithLegacy(table.baseTable(), snapshotId, BLOB_TYPE_OPTIMIZED_SEQUENCE);
+  }
+
+  public static StructLikeMap<Long> readBaseOptimizedTime(KeyedTable table) {
+    return readWithLegacy(table.baseTable(), null, BLOB_TYPE_BASE_OPTIMIZED_TIME);
+  }
+
+  public static StructLikeMap<Long> readBaseOptimizedTime(KeyedTable table, long snapshotId) {
+    return readWithLegacy(table.baseTable(), snapshotId, BLOB_TYPE_BASE_OPTIMIZED_TIME);
+  }
+
+  private static StructLikeMap<Long> readWithLegacy(
+      UnkeyedTable table, Long snapshotId, String type) {
+    if (snapshotId == null) {
+      Snapshot snapshot = table.currentSnapshot();
+      if (snapshot == null) {
+        return readLegacyPartitionProperties(table, type);
+      } else {
+        snapshotId = snapshot.snapshotId();
+      }
+    }
+    StructLikeMap<Long> result = readFromStatisticsFile(table, snapshotId, type);
+    return result != null ? result : readLegacyPartitionProperties(table, type);
+  }
+
+  private static StructLikeMap<Long> readFromStatisticsFile(
+      UnkeyedTable table, long snapshotId, String type) {
+    Snapshot snapshot = table.snapshot(snapshotId);
+    Preconditions.checkArgument(snapshot != null, "Snapshot %s not found", snapshotId);
+    List<StatisticsFile> statisticsFiles =
+        StatisticsFileUtil.findLatestValidStatisticsFiles(
+            table, snapshot.snapshotId(), StatisticsFileUtil.containsBlobOfType(type));
+    if (!statisticsFiles.isEmpty()) {
+      if (statisticsFiles.size() != 1) {
+        throw new IllegalStateException(
+            "There should be only one statistics file for blob type " + type);
+      }
+      List<StructLikeMap<Long>> result =
+          StatisticsFileUtil.reader(table)
+              .read(
+                  statisticsFiles.get(0),
+                  type,
+                  StatisticsFileUtil.createPartitionDataSerializer(table.spec(), Long.class));
+      if (result.size() != 1) {
+        throw new IllegalStateException(
+            "There should be only one partition data in statistics file for blob type " + type);
+      }
+      return result.get(0);
+    } else {
+      return null;
+    }
+  }
+
+  private static StructLikeMap<Long> readLegacyPartitionProperties(
+      UnkeyedTable table, String type) {
+    // to be compatible with old Amoro version 0.5.0 which didn't use puffin file and stored the
```

Comment:
Nit: Comments should start with a capital letter.

Reply:
Spotless automatically wraps lines at 100 characters, but you can make some adjustments for better readability here. Additionally, you can adjust the line length in your local IDEA to match the behavior of Spotless.